website/blog/2024-11-12-relational-vs-document-database-comparison.md

```diff
@@ -0,0 +1,196 @@
+  It consists of rows (elements or records) and columns (attributes) of data that are stored in multiple tables.
+- **ACID Compliance**: Transactions in relational databases are often ACID-compliant (Atomicity, Consistency, Isolation, Durability).
+  ACID is crucial for applications where data accuracy and integrity are essential, such as financial transactions.
+- **Complex Querying**: SQL provides powerful capabilities for complex querying, JOINs, and aggregation.
+  This is why they excel in use cases where advanced data manipulation and multi-table queries are required.
+- **Vertical Scalability**: They typically scale vertically by adding more resources to a single server rather than spreading across multiple servers.
+
+## What is a document database?
+
+With the rise of internet applications in the 2000s, the need to handle unstructured and semi-structured data became more pronounced.
+
+Traditional relational databases encountered limitations with unstructured data and scalability, particularly in distributed environments.
+This led to the development of NoSQL databases, including document-oriented databases like MongoDB, which was released in 2009.
+
+A document database or document-oriented database stores data in a flexible, semi-structured format called documents.
+They are usually encoded in BSON, XML, or JSON format.
+
+Unlike relational databases with their rigid table-based structure, document databases can include nested structures and varied fields.
+These systems were designed to meet the demands of rapidly growing web applications, where data structures often evolve quickly and scale is essential.
+
+Let's consider a social media application where user profiles, posts, and comments need to be stored.
+In this case, each profile or post might have nested information like user details, likes, and comments.
+Using a document database allows the application to store these complex structures in single unstructured documents that could take any form or data types.
+
+[Learn more about document databases](https://blog.ferretdb.io/document-databases-definition-features-use-cases/).
+
+## Key features of document databases
+
+- **Schema Flexibility**: No predefined schema is needed.
+  Fields can be added or modified as the application evolves.
+- **Data Encapsulation**: Each document encapsulates data as a single entity, reducing the need for JOIN operations, which are common in relational databases.
+- **Horizontal Scalability**: Document databases are typically built to scale horizontally, making them ideal for distributed systems and high-availability applications.
+- **High Availability**: Built with replication and sharding in mind, document databases are often easier to scale and distribute globally.
+
+## Comparing document databases and relational databases
+
+Here's a quick comparison of the characteristics of document and relational databases:
+
+```markdown
```

Comment:
But that's already a markdown file. Backticks prevent the table from being rendered correctly